iceprod/server/data/www_templates/job_detail.html

```diff
@@ -25,9 +25,9 @@
 
 {% if (current_user) %}
 <div class="action">
-    <button onclick='call("reset_jobs");'>Reset</button>
-    <button onclick='call("hard_reset_jobs");'>Hard Reset</button>
-    <button onclick='call("suspend_jobs");'>Suspend</button>
+    <button onclick='set_jobs_status(["{{ job["job_id"] }}"], "reset", "{{ passkey }}", filter=["&task_status=failed", "&task_status=suspended"]);'>Reset</button>

```

Comment:
If I'm reading this right, you are trying to only reset when the task status is failed or suspended.  This is incorrect, as a reset should be allowed for any state other than complete.

Reply:
What's the difference between reset and hard reset then?

Reply:
Hard reset can reset complete tasks (so it goes back to the beginning, starts over)

Reply:
OK, I misunderstood what needs to be done. This will need to re-done.